databricks_cli/dbfs/api.py

```diff
@@ -97,20 +119,37 @@ def file_exists(self, dbfs_path, headers=None):
             raise e
         return True
 
+    @retry_429
     def get_status(self, dbfs_path, headers=None):
         json = self.client.get_status(dbfs_path.absolute_path, headers=headers)
         return FileInfo.from_json(json)
 
+    @retry_429
+    def create(self, dbfs_path, overwrite, headers):
+        return self.client.create(dbfs_path, overwrite, headers=headers)
+
+    @retry_429
+    def add_block(self, handle, contents, headers):
+        self.client.add_block(handle, b64encode(contents).decode(), headers=headers)
+
+    @retry_429
+    def close(self, handle, headers):
+        self.client.close(handle, headers=headers)
+
     def put_file(self, src_path, dbfs_path, overwrite, headers=None):
-        handle = self.client.create(dbfs_path.absolute_path, overwrite, headers=headers)['handle']
+        handle = self.create(dbfs_path, overwrite, headers=headers)['handle']
         with open(src_path, 'rb') as local_file:
             while True:
                 contents = local_file.read(BUFFER_SIZE_BYTES)
                 if len(contents) == 0:
                     break
                 # add_block should not take a bytes object.
-                self.client.add_block(handle, b64encode(contents).decode(), headers=headers)
```

Comment:
For this particular case, I think it's better to do `b64encode(contents).decode()` here and not move it to the new `add_block` method, because here it makes sense, while in the new method the context is lost, you don't know what the type of `contents` is and why you need to decode it.